tsdb/engine/tsm1/engine.go

```diff
@@ -2095,24 +2095,28 @@ func (e *Engine) compact(wg *sync.WaitGroup) {
 			}
 
 			// Find our compaction plans
-			level1Groups := e.CompactionPlan.PlanLevel(1)
-			level2Groups := e.CompactionPlan.PlanLevel(2)
-			level3Groups := e.CompactionPlan.PlanLevel(3)
-			level4Groups := e.CompactionPlan.Plan(e.LastModified())
-			atomic.StoreInt64(&e.stats.TSMOptimizeCompactionsQueue, int64(len(level4Groups)))
+			level1Groups, len1 := e.CompactionPlan.PlanLevel(1)
+			level2Groups, len2 := e.CompactionPlan.PlanLevel(2)
+			level3Groups, len3 := e.CompactionPlan.PlanLevel(3)
+			level4Groups, len4 := e.CompactionPlan.Plan(e.LastModified())
+			atomic.StoreInt64(&e.stats.TSMOptimizeCompactionsQueue, len4)
 
 			// If no full compactions are need, see if an optimize is needed
 			if len(level4Groups) == 0 {
```

Comment:
Double-checking: should this be checking `len4` now?

Reply:
No, we want the new `len?` parameters for display only. There should be no functional changes to operations, so we continue to respect the acquisition of locks on the files to be compacted.